pkg/controllers/synccommon/applyconfiguration.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2025 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package synccommon
+
+import (
+	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	machinev1applyconfigs "github.com/openshift/client-go/machine/applyconfigurations/machine/v1beta1"
+)
+
+// syncObjApplyConfiguration is an apply configuration for objects managed by
+// the sync controller. This is currently MAPI Machine and MachineSet.
+type syncObjApplyConfiguration[objPT any, statusPT syncStatusApplyConfiguration[statusPT]] interface {
+	WithStatus(statusPT) objPT
+	WithResourceVersion(string) objPT
+}
+
+// syncObjApplyConfigurationP asserts that a syncObjApplyConfiguration is a pointer to a specific concrete type.
```

Comment:
my head's hurting a little (I think this is just generics) - why do we need to assert we have pointers to concrete types?